express-serve-static-core/express-serve-static-core.d.ts

```diff
@@ -66,11 +73,11 @@ declare module "express-serve-static-core" {
             * @param name
             * @param fn
             */
-        param(name: string, handler: RequestParamHandler): T;
-        param(name: string, matcher: RegExp): T;
-        param(name: string, mapper: (param: any) => any): T;
+        param(name: string | string[], handler: RequestParamHandler): this;
+        param(name: string | string[], matcher: RegExp): this;
```

Comment:
If not return this, then it will be a generic T.
for example, in old solution, there is  IRouter<T>, and interface Router extends IRouter<Router>
interface Application extends IRouter<Application>